tests/std/tests/P0896R4_ranges_alg_move/test.cpp

```diff
@@ -18,66 +20,63 @@ struct int_wrapper {
-        assert(result.in == basic_borrowed_range{input}.end());
-        assert(result.out == basic_borrowed_range{output}.end());
-        assert(ranges::equal(output, input));
-    }
-    {
-        int output[] = {-2, -2, -2};
-        basic_borrowed_range wrapped_input{input};
-        auto result = move(wrapped_input.begin(), wrapped_input.end(), basic_borrowed_range{output}.begin());
-        STATIC_ASSERT(same_as<decltype(result),
-            move_result<iterator_t<basic_borrowed_range<int const>>, iterator_t<basic_borrowed_range<int>>>>);
-        assert(result.in == wrapped_input.end());
-        assert(result.out == basic_borrowed_range{output}.end());
-        assert(ranges::equal(output, input));
-    }
-    {
-        int_wrapper input1[3]        = {13, 55, 1234};
-        int const expected_output[3] = {13, 55, 1234};
-        int_wrapper actual_output[3] = {-2, -2, -2};
-        basic_borrowed_range wrapped_input{input1};
-        auto result = move(wrapped_input.begin(), wrapped_input.end(), basic_borrowed_range{actual_output}.begin());
-        assert(result.in == wrapped_input.end());
-        assert(result.out == basic_borrowed_range{actual_output}.end());
-        for (int i = 0; i < 3; ++i) {
-            assert(input1[i].val == -1);
-            assert(actual_output[i].val == expected_output[i]);
+struct instantiator {
+    static constexpr int input[3]                   = {13, 53, 12435};
+    static constexpr int_wrapper input1[3]          = {13, 55, 12345};
+    static constexpr int_wrapper expected_output[3] = {13, 55, 12345};
+    static constexpr int_wrapper expected_input[3]  = {-1, -1, -1};
+
+    template <class Read, class Write>
+    static constexpr void call() {
+        using ranges::move, ranges::move_result, ranges::iterator_t;
+        {
+            int output[3] = {-2, -2, -2};
+            Read wrapped_input{input};
+
+            auto result = move(wrapped_input, Write{output});
+            STATIC_ASSERT(same_as<decltype(result), move_result<iterator_t<Read>, Write>>);
+            assert(result.in == wrapped_input.end());
+            assert(result.out.peek() == Write{output + 3});
+            assert(ranges::equal(output, input));
         }
-    }
-}
+        {
+            int output[]                   = {-2, -2, -2};
+            int_wrapper expected_output[3] = input1;
+            Read wrapped_input{input};
 
-int main() {
-    STATIC_ASSERT((smoke_test(), true));
-    smoke_test();
-}
+            auto result = move(wrapped_input.begin(), wrapped_input.end(), Write{output});
+            STATIC_ASSERT(same_as<decltype(result), move_result<iterator_t<Read>, Write>>);
+            assert(result.in == wrapped_input.end());
+            assert(result.out.peek() == Write{output + 3});
+            assert(ranges::equal(output, input));
+        }
+        {
+            int_wrapper output[3] = {-2, -2, -2};
+            Read wrapped_input{input1};

```

Comment:
`Read`'s element type is `const int` (`Write`'s is `int`), and `input1` is a contiguous range of `const int_wrapper` (`output` is `int_wrapper`); this should be ill-formed. You'll need to either make all of the tests in `instantiator::call` be `int_wrapper` tests and change the element types on lines 80-81, or separate this case into a separate class and instantiate that class with a separate `test_in_write` call.

Reply:
Went to remove the int test as the wrapper_int tests are strictly more powerfull